exporters/prometheus/exporter.go

```diff
@@ -72,8 +81,11 @@ func (c *collector) Describe(ch chan<- *prometheus.Desc) {
 	metrics, err := c.reader.Collect(context.TODO())
 	if err != nil {
 		otel.Handle(err)
+		if err == metric.ErrReaderNotRegistered {
```

Comment:
This was needed because at the creation of the exporter, we register it to the `prometheus.Registry` but that reader was not registered at the `MeterProvider` yet. 

So we would have a call with an empty `ResourceMetrics` and was caching that on the `target_info` variable.

Should we really try to export things if there was a error? For now, I'll skip just the `ErrReaderNotRegistered` because that is guaranteed to not have data.

Reply:
I think this is a bug. The code on `main` is complaining about this currently and it means that no metrics at all are exposed if default reader is used.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/3291